### PR TITLE
Support JVM system proxy in mvn artifacts listing

### PR DESCRIPTION
**Pull request by rinconjc**
Before submitting a PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [ x] The commits are consistent with our [contribution guidelines](CONTRIBUTING.md)
- [x] You've added tests (if possible) to cover your change(s)
- [ x] All tests are passing (run `lein do clean, test`)
- [x] Code inlining with mranderson works and tests pass with inlined code (run `./build.sh install` -- takes a long time)
- [x] You've updated the changelog (if adding/changing user-visible functionality)
- [x] You've updated the readme (if adding/changing user-visible functionality)

Thanks!

### CODE REVIEW

File: src/refactor_nrepl/artifacts.clj [hunk 1]
```diff
@@ -17,6 +17,14 @@
 (defonce artifacts (atom {} :meta {:last-modified nil}))
 (def millis-per-day (* 24 60 60 1000))
 
+(defn- get-proxy-opts
+  "Generates proxy options from JVM properties for httpkit-client "
+  []
+  (if-let [proxy-host (some #(System/getProperty %) ["http.proxyHost" "https.proxyHost"])]
```
Comment (benedekfazekas) [inline src/refactor_nrepl/artifacts.clj]: i would prefer a `when-let` here

File: src/refactor_nrepl/artifacts.clj [hunk 2]
```diff
@@ -17,6 +17,14 @@
...
+    {:proxy-host proxy-host
+     :proxy-port (some->> ["http.proxyPort" "https.proxyPort"]
```
Comment (benedekfazekas) [inline src/refactor_nrepl/artifacts.clj]: the code prefers `http` proxy over `https`. perhaps the other way around would be a better idea?

**Comment by rinconjc** [comment]
Updated according to your comments. Please review, thanks

**Comment by benedekfazekas** [comment]
thanks for the quick turn around. 

can you please update the changelog and for this actually the readme too. ta!

**Comment by rinconjc** [comment]
Updated docs. Please review

**Comment by benedekfazekas** [comment]
thanks!

**Comment by expez** [comment]
Great work @rinconjc! :+1:

**Comment by benedekfazekas** [comment]
new snapshot is on clojars

**Pull request merged**